### PR TITLE
Add deprecation check for processors

### PR DESCRIPTION
**Pull request by jasontedor**
The processors setting is deprecated. This commit adds a deprecation check for the use of the processors setting.

Relates #45855 
Relates #45905

### CODE REVIEW

File: x-pack/plugin/deprecation/src/main/java/org/elasticsearch/xpack/deprecation/NodeDeprecationChecks.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    static DeprecationIssue checkProcessors(final Settings settings , final PluginsAndModules pluginsAndModules) {
+        if (EsExecutors.PROCESSORS_SETTING.exists(settings) == false) {
+            return null;
+        }
+        final String message = String.format(
+            Locale.ROOT,
+            "setting [%s] is deprecated in favor of setting [%s]",
+            EsExecutors.PROCESSORS_SETTING.getKey(),
+            EsExecutors.NODE_PROCESSORS_SETTING.getKey());
+        final String url = "https://www.elastic.co/guide/en/elasticsearch/reference/master/breaking-changes-8.0.html#remove-processors";
```
Comment (rjernst) [inline x-pack/plugin/deprecation/src/main/java/org/elasticsearch/xpack/deprecation/NodeDeprecationChecks.java]: This link refers to master, which may be different when someone sees this setting (for example once 7.x is in maintenance mode)? I think we need a better permalink if we are to include this, but haven't we avoided links in exception messages in the past for this exact reason?

Comment (jasontedor) [reply x-pack/plugin/deprecation/src/main/java/org/elasticsearch/xpack/deprecation/NodeDeprecationChecks.java]: Let me make a docs push to add a deprecation note to the 7.4 docs which can be used as a permalink.

Comment (rjernst) [reply x-pack/plugin/deprecation/src/main/java/org/elasticsearch/xpack/deprecation/NodeDeprecationChecks.java]: I've just realized we have the same issue for the existing DeprecationIssue instances, which we should probably (separately from this PR) fix.

Comment (jasontedor) [reply x-pack/plugin/deprecation/src/main/java/org/elasticsearch/xpack/deprecation/NodeDeprecationChecks.java]: @gwbrown Can you address that?

Comment (jasontedor) [reply x-pack/plugin/deprecation/src/main/java/org/elasticsearch/xpack/deprecation/NodeDeprecationChecks.java]: @rjernst I pushed a permalink that is currently broken, but will be fixed by the time the 7.4 release is in published.

**Review by rjernst — APPROVED** [review]
LGTM

**Pull request merged**